JWT 리프레시토큰 리팩토링

happymink opened the pull request:
# 내용
- 만료된 JWT Refresh를 관리하기 위한 기능을 추가합니다.
- Member Entity에 Basket 연관관계 매핑을 추가합니다.

# 주요 리팩토링
- Refresh 토큰을 통해 AccessToken을 재발급 할 때마다 `RefreshTokenRepository`의 `deleteRefreshTokensByExpiryDateBefore`를 통해 만료된 시간의 RefreshToken을 제거하는 기능, 이에 따른 테스트를 추가합니다.

# 추가적인 내용
- 이외에도 적용하진 않았지만 Spring Batch Job으로 주기적인 관리를 할 수 있을 것 같습니다.

happymink committed:
[refactor] Member에 Basket 연관관계 설정

happymink committed:
[refactor] jwtService에 만료된 Refresh 토큰 제거 기능 추가

happymink committed:
[test] 만료된 RefreshToken 제거 test 추가

happymink committed:
[refactor] signUp 과정에서 객체 생성, DB save 단계를 분리했습니다.

happymink committed:
[refactor] id를 포함하는 Member 생성자에서 id 항목을 제거합니다.

happymink commented:
![image](https://github.com/Yanolza-Miniproject/backend/assets/71982555/94bece33-4ab3-4c48-91a3-ab1d5cffc47c)

멘토님께서 남겨주신 리뷰에 대해서 Entity의 생성자, Builder 패턴을 언제 어떻게 사용해야 하는지 감이 잘 안오는 것 같습니다. 
생성자의 Id를 닫는 이유는 DB에서 auto increment로 Id 값을 지정해줘서 그런걸까요?

meena2003 commented on src/main/java/com/miniproject/domain/member/entity/Member.java:
```diff
@@ -20,6 +21,10 @@ public class Member {
+
     @Id
     @GeneratedValue(strategy = GenerationType.IDENTITY)
     private Long id;
```
이건 제 생각인데, 기본키가 가장 맨 위로 올라가면 어떨까 싶어요. 가장 기본이 되는 속성이라고 생각되서요!

meena2003 commented on src/main/java/com/miniproject/domain/member/entity/Member.java:
```diff
@@ -35,16 +40,14 @@ public class Member {
         this.orders = orders;
-
     }
 
```
별로 중요한 내용은 아닌것 같지만, 이 빈칸도 없애는 게 더 보기 좋지 않을까요? ㅎㅎ

meena2003 commented on src/main/java/com/miniproject/domain/member/service/MemberService.java:
롬복의 @RequiredArgsConstructor를 사용하지 않고 생성자를 사용하신 이유가 있나요?

meena2003 commented on src/main/java/com/miniproject/domain/member/service/MemberService.java:
```diff
@@ -29,12 +29,14 @@ public MemberService(MemberRepository memberRepository, PasswordEncoder password
     public Long signUp(SignUpRequest request) {
```
메소드에 @Transitional 애노테이션이 없어도 괜찮나요? 서비스 계층에선 하나의 메소드가 동일한 트랜잭션 내부에서 동작해야 한다고 알고 있어서요

meena2003 reviewed (approved):
어려운 내용이지만 잘 구현하신것 같습니다. 수고하셨습니다!

happymink committed:
[refactor] MemberService 트랜잭셔널 어노테이션 추가

happymink committed:
[refactor] MemberEntity 필드 순서 정리

junmo95 reviewed (approved):
👍

KwonJuHwan commented on src/main/java/com/miniproject/domain/member/service/MemberService.java:
```diff
@@ -26,17 +27,21 @@ public MemberService(MemberRepository memberRepository, PasswordEncoder password
+        return memberRepository.save(member).getId();
     }
 
+    @Transactional(readOnly = true)
```
말씀해주신 것처럼, MemberService 단의 default 를 `@Transactional(readOnly= true)`로 지정하고, 트랜잭션이 필요한 환경 즉, DB에 영향이 가는 메소드만 `@Transactional`을 붙이는게 더 좋아보입니다!

happymink committed:
[refactor] Order, Payment 로직 수정

Motivation:

 - Member Entity의 생성자에 Id를 제외함에따라 Test코드를 수정하는 과정에서 Member의 getId 로직을 getEmail로 수정합니다.

 - Test에 의해 기존 로직을 수정하면 안되지만, 임의로 일단 수정해서 올립니다..!

Modification:

 - OrderService, PaymentService 수정

happymink committed:
[refactor] Member Entity 생성자 수정으로 인한 전체 테스트 수정

Motivation:

 - Member 생성자에서 Id 항목을 제외함으로써 기존 테스트의 Member id 요소를 제거합니다.

Modification:

 - Member Build 과정에서 .id() 제거

happymink commented:
Member Entity 생성자에 id 항목을 제거하고 그에 따른 Test 코드, Member를 Build하는 과정을 임의로 수정했습니다. 
Test Code를 위해 서비스 코드를 수정하면 안되지만.. OrderService, PaymentService에서 Member.id를 Member.email로 변경했습니다. 확인하시고 수정 부탁드립니다..!
